var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -337,11 +406,42 @@ def find_optional_library(name, prefix):
                 'SUPERLU_OPT=-I%s -I%s' %
                 (spec['superlu-dist'].prefix.include,
                  spec['parmetis'].prefix.include),
-                'SUPERLU_LIB=-L%s -L%s -lsuperlu_dist -lparmetis %s' %
-                (spec['superlu-dist'].prefix.lib,
-                 spec['parmetis'].prefix.lib,
+                'SUPERLU_LIB=%s %s' %
+                (ld_flags_from_dirs([spec['superlu-dist'].prefix.lib,
+                                     spec['parmetis'].prefix.lib],
+                                    ['superlu_dist', 'parmetis']),
                  ld_flags_from_library_list(lapack_blas))]
 
+        if '+strumpack' in spec:
+            strumpack = spec['strumpack']
+            sp_opt = ['-I%s' % strumpack.prefix.include]
+            sp_lib = [ld_flags_from_library_list(strumpack.libs)]
+            # Parts of STRUMPACK use fortran, so we need to link with the
+            # fortran library and also the MPI fortran library:
+            if '~shared' in strumpack:
+                if os.path.basename(env['FC']) == 'gfortran':
+                    sp_lib += ['-lgfortran']
+                if '^mpich' in strumpack:
+                    sp_lib += ['-lmpifort']
+                elif '^openmpi' in strumpack:
+                    sp_lib += ['-lmpi_mpifh']
```

Comment:
@tgamblin, is there a better way to handle these fortran dependecies?

Reply:
Sorry, didn't see this in the deluge of libceed notifications.  Not right now there isn't.  But compilers-as-dependencies are coming soon, and we should be able to support mixed compiler builds way better then.

Reply:
OK, we can revisit later. Thanks!